Improve documentation about getting default output stream to a device

CleanCut opened the pull request:
Resolves #355 

/cc @grzi I ran into this as well!

est31 reviewed (changes requested):
Thanks! Could you use intra-doc-links instead of linking to the html? https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html

est31 commented:
Also not a native speaker but I think it should be "to *the* system's default sound device"

CleanCut commented:
> Also not a native speaker but I think it should be "to _the_ system's default sound device"

Good catch, fixed!

> Thanks! Could you use intra-doc-links instead of linking to the html?

Excellent, I've been wanting an excuse to try out the new doc linking method.

✅ It is done. Take another look!

I may have ended up going a bit overboard. 😊 I started by updating just my links, and then I went and fixed all the other links, and then I noticed the examples were out of date and broken so I fixed those, and I saw some wording that could be fixed... Anyway, I ended up refreshing quite a bit of the documentation.

- [x] All the links work for me when generated locally
- [x] The doctests all work for me locally.
  - Should we restore the `no_run` on the doctest that plays sound?  I prefer always running doctests, but I can understand why one would not want to if sound-less systems like CI crashed.

est31 commented on src/lib.rs:
````diff
@@ -1,68 +1,77 @@
 //! For example, here is how you would play an audio file:
 //!
-//! ```no_run
+//! ```
````
Travis CI failed... maybe add back the no_run here?

CleanCut replied:
✅ Sure thing

CleanCut commented:
Travis is now happy once again. Though I think server farms should include sound cards and pipe all the sound to the overhead public-address system as entertainment for those in the building. 😆
